File: backend/models/playbook.py
from sqlalchemy import Column, Integer, String, DateTime, JSON
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from ..database import Base

class Playbook(Base):
    __tablename__ = "playbooks"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    description = Column(String)
    version = Column(String, nullable=False)
    steps = Column(JSON, nullable=False)  # List of steps with their configurations
    parameters = Column(JSON)  # Expected parameters and their types
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())

    # Relationships
    tasks = relationship("Task", back_populates="playbook") 

File: src/types/playbook.ts
import { Task } from './task';

export type PlaybookStatus = 'pending' | 'running' | 'completed' | 'failed';

export interface Playbook {
  id: string;
  name: string;
  steps: Task[];
  assignedAgentId?: string;
  status: PlaybookStatus;
  createdAt: string;
  updatedAt: string;
  completedAt?: string;
  error?: string;
} 